ddl/ingest/backend.go

```diff
@@ -133,38 +133,47 @@ func (bc *litBackendCtx) Flush(indexID int64, force bool) (flushed, imported boo
 		return false, false, dbterror.ErrIngestFailed.FastGenByArgs("ingest engine not found")
 	}
 
-	bc.diskRoot.UpdateUsage()
-	shouldImport := bc.diskRoot.ShouldImport()
-	shouldFlush := force ||
-		shouldImport ||
-		time.Since(bc.timeOfLastFlush.Load()) >= bc.updateInterval
+	shouldFlush, _ := bc.ShouldSync(force)
+	if !shouldFlush {
+		return false, false, nil
+	}
+	ei.flushLock.Lock()
+	defer ei.flushLock.Unlock()
+	// Check for the second time to avoid duplicate import.
+	shouldFlush, shouldImport := bc.ShouldSync(force)
```

Comment:
I do not think the second-round check is necessary.

Reply:
Since we only allow one writelock at one time, can we add one atomic bool to show whether the lock has been granted currently.